src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializerOptions.Caching.cs

```diff
@@ -86,7 +112,7 @@ private void InitializeCachingContext()
         internal sealed class CachingContext
         {
             private readonly ConcurrentDictionary<Type, JsonConverter> _converterCache = new();
-            private readonly ConcurrentDictionary<Type, JsonTypeInfo> _jsonTypeInfoCache = new();
+            private readonly ConcurrentDictionary<Type, JsonTypeInfo?> _jsonTypeInfoCache = new();
```

Comment:
What does caching `null` JsonTypeInfo represent?

Reply:
null means that type can't be resolved (minus special case for JsonSerializerContext which starts behaving differently once you call certain APIs)

Reply:
So basically caching potential `null` results from the underlying resolver?

Reply:
correct